crest/src/main/java/com/hartwig/hmftools/crest/CrestApplication.java

```diff
@@ -0,0 +1,205 @@
+    {
+        logVersion();
+
+        ConfigBuilder configBuilder = new ConfigBuilder();
+        registerConfig(configBuilder);
+        configBuilder.checkAndParseCommandLine(args);
+
+        CrestApplication crestApplication = new CrestApplication(configBuilder);
+        crestApplication.logParams();
+        crestApplication.run();
+    }
+
+    public static void logVersion()
+    {
+        final VersionInfo version = new VersionInfo("crest.version");
+        LOGGER.info("Crest version: {}", version.version());
+    }
+
+    void run() throws IOException
+    {
+        String rnaAnnotatedGermlineVcf = PurpleCommon.purpleGermlineVcfFile(purpleDir, sampleId);
+        LOGGER.info("Checking file: {}", rnaAnnotatedGermlineVcf);
+        double supportRatio = computeRnaSupportRatio(rnaAnnotatedGermlineVcf);
+
+        String outputFilename;
+        if(supportRatio < acceptanceRatio)
+        {
+            LOGGER.error("Check failed, ratio of supported reads is below threshold");
+            outputFilename = getOutputFilename(false);
+        }
+        else
+        {
+            LOGGER.info("Check succeeded");
+            outputFilename = getOutputFilename(true);
+        }
+
+        if(!doNotWriteFile)
+        {
+            LOGGER.info("Writing file: {}", outputFilename);
+            new FileOutputStream(outputFilename).close();
+        }
+    }
+
+    private String getOutputFilename(boolean success)
+    {
+        String extension = success ? ".CrestCheckSucceeded" : ".CrestCheckFailed";
+        return (outputDir == null ? "" : outputDir) + sampleId + extension;
+    }
+
+    public void logParams()
+    {
+        LOGGER.info("purpleDir: {}", purpleDir);
+        LOGGER.info("outputDir: {}", outputDir);
+        LOGGER.info("sampleId: {}", sampleId);
+        LOGGER.info("sampleToCheck: {}", sampleToCheck);
+        LOGGER.info("minTotalReads: {}", minTotalReads);
+        LOGGER.info("minRnaReads: {}", minRnaReads);
+        LOGGER.info("acceptanceRatio: {}", acceptanceRatio);
+        LOGGER.info("doNotWriteFile: {}", doNotWriteFile);
+    }
+
+    public double computeRnaSupportRatio(@NotNull String vcfFilename) throws IOException
+    {
+        int supported = 0;
+        var total = 0;
+
+        try(AbstractFeatureReader<VariantContext, LineIterator> reader = getFeatureReader(vcfFilename, new VCFCodec(), false))
+        {
+            final VCFHeader header = (VCFHeader) reader.getHeader();
+            if(!sampleInFile(sampleToCheck, header))
+            {
+                throw new RuntimeException("Sample " + sampleToCheck + " not found in file " + vcfFilename);
+            }
+
+            for(VariantContext context : reader.iterator())
+            {
+                VariantContextDecorator decorator = new VariantContextDecorator(context);
+
+                if(decorator.filter().equals("PASS") && decorator.type() == VariantType.SNP && !decorator.gene().isEmpty())
```

Comment:
Cleaner to reuse `isPass` function on `VariantContextDecorator`?

Reply:
Didn't notice that one, thanks!